### ISSUE TITLE
Jetson nano has problem with start up from mmc 

### ISSUE THREAD

**Issue by Wlad79** [OE4T/tegra-demo-distro#288]
Hello, 

we want to use Jetson nano on branch dunfell-l4t-r32.4.3 and have got followed message at u-boot start, kernel is not loaded because my rootfs is on partition 1 and u-boot need it on 0 regarding followed message:

`
U-Boot 2020.04 (Jan 05 2022 - 19:59:34 +0000)

SoC: tegra210
Model: NVIDIA Jetson Nano Developer Kit
Board: NVIDIA P3450-0002
DRAM:  4 GiB
MMC:   sdhci@700b0000: 1, sdhci@700b0600: 0
Loading Environment from MMC... *** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
Net:   No ethernet found.
Hit any key to stop autoboot:  0 
switch to partitions #0, OK
mmc0(part 0) is current device
Scanning mmc 0:1...
starting USB...
Bus usb@7d000000: tegrausb: Invalid dr_mode 2 for host mode
probe failed, error -1
No working controllers found
No ethernet found.
<...>
No ethernet found.
Config file not found
starting USB...
Bus usb@7d000000: tegrausb: Invalid dr_mode 2 for host mode
probe failed, error -1
No working controllers found
No ethernet found.
No ethernet found.
Tegra210 (P3450-0002) # ls mmc 0:1
<DIR>       4096 .
<DIR>       4096 ..
<DIR>      16384 lost+found
<DIR>       4096 bin
<DIR>       4096 boot
<DIR>       4096 dev
<DIR>       4096 etc
<DIR>       4096 home
<DIR>       4096 lib
<DIR>       4096 media
<DIR>       4096 mnt
<DIR>       4096 proc
<DIR>       4096 run
<DIR>       4096 sbin
<DIR>       4096 sys
<DIR>       4096 tmp
<DIR>       4096 usr
<DIR>       4096 var
Tegra210 (P3450-0002) # ls mmc 0:0
** Unrecognized filesystem type **
`

**Comment by Wlad79** [comment]
`Tegra210 (P3450-0002) # mmc part

Partition Map for MMC device 0  --   Partition Type: EFI

Part    Start LBA       End LBA         Name
        Attributes
        Type GUID
        Partition GUID
  1     0x00000022      0x01c00021      "APP"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   6f849c61-d5e9-4850-a6dd-d36e195edd17
  2     0x01c00022      0x01c00821      "DTB"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   1a5503f3-ba87-46df-9b0e-e40916e53139
  3     0x01c00822      0x01c03821      "TOS"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   558dd825-2920-42b4-ba3e-0261b1e59915
  4     0x01c03822      0x01c038c1      "EKS"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   1d1a72d9-608a-4bc4-be7a-2a04f51d4d65
  5     0x01c038c2      0x01c238a1      "LNX"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   1b235e12-921c-4b19-bc93-c3725265517e
  6     0x01c238a2      0x01c240a1      "DTB-1"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   710fc174-7479-4f03-8ff4-e20d66525d6a
  7     0x01c240a2      0x01c270a1      "TOS-1"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   37689b2d-8e70-41d2-a4df-bf4060c47821
  8     0x01c270a2      0x01c27141      "EKS-1"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   31722c39-fda1-42b0-b2dc-f43e9ac8f620
  9     0x01c27142      0x01c47121      "LNX-1"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   1b01d38a-ba98-4dc8-b326-d4387dd75635
 10     0x01c47122      0x01c472a1      "BMP"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   44a8751f-34ce-42b8-94bc-886e454d361a
 11     0x01c472a2      0x01c474a1      "RP4"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   656c5dee-fb8e-4f8a-b632-d02fc7d08602
 12     0x01c474a2      0x01c66ca1      "recovery"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   5b4f5c18-ad34-43fa-bcee-d3672bba7276
 13     0x01c66ca2      0x01c670a1      "recovery-dtb"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   0f143f51-81f9-4a97-bd1f-c474c5002400
 14     0x01c670a2      0x01c672a1      "kernel-bootctrl"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   099af672-140d-42a7-ac53-816a9f910341
 15     0x01c672a2      0x01c674a1      "kernel-bootctrl_b"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   6479a27d-3250-4b41-bf55-7c62b6ceeb15
 16     0x01c674a2      0x01cfd4a1      "RECROOTFS"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   5df22ff1-32b2-4171-9097-e2367c03f478
 17     0x01cfd4a2      0x01d59fde      "UDA"
        attrs:  0x0000000000000000
        type:   ebd0a0a2-b9e5-4433-87c0-68b6b72699c7
        guid:   4f39ed4a-be03-4fb6-b9da-4a2e6a62e213`

**Comment by Wlad79** [comment]
Hello, 
can someone say me what should be done for eliminating the error with boot from mmc in u-boot?

I really appreciate the help on this topic.

**Comment by madisongh** [comment]
I just tried a build myself and loaded it up, and I'm not having any problems booting:
```
U-Boot 2020.04 (Jan 05 2022 - 19:59:34 +0000)

SoC: tegra210
Model: NVIDIA Jetson Nano Developer Kit
Board: NVIDIA P3450-0002
DRAM:  4 GiB
MMC:   sdhci@700b0000: 1, sdhci@700b0600: 0
Loading Environment from MMC... *** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
Net:   No ethernet found.
Hit any key to stop autoboot:  0 
switch to partitions #0, OK
mmc0(part 0) is current device
Scanning mmc 0:1...
Found /boot/extlinux/extlinux.conf
Retrieving file: /boot/extlinux/extlinux.conf
224 bytes read in 17 ms (12.7 KiB/s)
1:	OE4Tegra Demonstration Distro
Retrieving file: /boot/extlinux/../initrd
2096405 bytes read in 67 ms (29.8 MiB/s)
Retrieving file: /boot/extlinux/../Image
37335048 bytes read in 876 ms (40.6 MiB/s)
```
This is with the latest on the `dunfell-l4t-r32.4.3` branch here, built on a Ubuntu 20.04 host.

It looks like with your setup, U-Boot isn't finding the `/boot/extlinux/extlinux.conf` file on partition `mmc 0:1`.  Did you make changes that might have affected the content of the rootfs partition?  From U-Boot, can you manually run the boot yourself?
```
Tegra210 (P3450-0002) # devnum=0; distro_bootpart=1; sysboot mmc ${devnum}:${distro_bootpart} any ${scriptaddr} /boot/extlinux/extlinux.conf
```

**Comment by Wlad79** [comment]
@madisongh thanks for fast response, 
I'm using ubuntu18.04LTS as host. But I can remember that I have tested it already on ubuntu22.04 with same result like ubuntu18.04LTS. In both builds I have nothing that effects u-boot and kernel in wrong method. I have tested in early yocto version with DSI  functions for kernel, what was in that version okay.

I will test the build on ubuntu20.04 until middle next week.
